packages/terra-time-input/src/TimeInput.jsx

```diff
@@ -460,7 +460,18 @@ class TimeInput extends React.Component {
       if (TimeUtil.validateTime(currentTimeValue, this.props.showSeconds)) {
         if (minute === '0' || minute === '00') {
           minute = '59';
-          hour = TimeUtil.decrementHour(hour);
+          if (hour === '0' || hour === '00') {
+            hour = '23';
+          } else {
+            if (variant === TimeUtil.FORMAT_12_HOUR && hour === '12') {
+              if (meridiem === this.anteMeridiem) {
+                meridiem = this.postMeridiem;
+              } else {
+                meridiem = this.anteMeridiem;
+              }
+            }
+            hour = TimeUtil.decrementHour(hour);
+          }
```

Comment:
We should add some tests for the changes in this file.

Reply:
Added here https://github.com/cerner/terra-framework/pull/1520/commits/23db2ef02289816b2cbf9619172dc497a918286f#diff-2bc93a8867c8555b5abe36e5b499d356c82131d59cb0390dcc5668fb928a8ba0R126